### PR TITLE
(fix) Allow opening legacy html forms on visit list

### PR DESCRIPTION
**Pull request by icrc-psousa**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This fix will allow editing encounters recorded on HTMLFormEntry from the visits list. Prior to this fix, if the user tried to open a legacy form, an error would occur.
The configuration used to check whether a form is legacy html or not is the same already in place for the "Forms & Notes" list.

## Screenshots
image

### CODE REVIEW

File: packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx [hunk 1]
```diff
@@ -43,11 +52,19 @@ type FilterProps = {
...
 };
 
-const VisitTable: React.FC<VisitTableProps> = ({ showAllEncounters, visits }) => {
+const VisitTable: React.FC<VisitTableProps> = ({ showAllEncounters, visits, patientUuid}) => {
   const visitCount = 20;
   const { t } = useTranslation();
+  const config = useConfig();
   const isTablet = useLayoutType() === 'tablet';
 
+  const [htmlFormEntryFormsConfig, setHtmlFormEntryFormsConfig] = React.useState<null | Array<HtmlFormEntryForm>>(null);
```
Comment (ibacher) [inline packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx:61]:
```suggestion
  const [htmlFormEntryFormsConfig, setHtmlFormEntryFormsConfig] = useState<Array<HtmlFormEntryForm> | undefined>();
```

File: packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx [hunk 2]
```diff
@@ -101,8 +118,15 @@ const VisitTable: React.FC<VisitTableProps> = ({ showAllEncounters, visits }) =>
     formName?: string,
     visitTypeUuid?: string,
   ) => {
-    formEntrySub.next({ formUuid, visitUuid, encounterUuid, visitTypeUuid });
-    launchPatientWorkspace('patient-form-entry-workspace', { workspaceTitle: formName });
+    const htmlForm = htmlFormEntryFormsConfig.find((form) => form.formUuid === formUuid);
```
Comment (ibacher) [inline packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx:121]:
```suggestion
    const htmlForm = htmlFormEntryFormsConfig?.find((form) => form.formUuid === formUuid);
```

File: packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx [hunk 3]
```diff
@@ -24,15 +24,24 @@ import {
...
+import {
+  formEntrySub,
+  launchPatientWorkspace,
+  launchStartVisitPrompt,
+  PatientChartPagination
+} from '@openmrs/esm-patient-common-lib';
 import { MappedEncounter } from '../visit-summary.component';
 import EncounterObservations from '../../encounter-observations';
 import styles from './visits-table.scss';
+import {HtmlFormEntryForm} from "@openmrs/esm-patient-forms-app/src/config-schema";
```
Comment (ibacher) [inline packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/visits-table/visits-table.component.tsx:37]:
```suggestion
import type {HtmlFormEntryForm} from "@openmrs/esm-patient-forms-app/src/config-schema";
```

**Review by ibacher — CHANGES_REQUESTED** [review]
Thanks for this. This doesn't look like it's properly formatted according to our standards (which should be happening on commit, but if you've disabled that hook, you do need to run it). A few other things to fix highlighted below.

File: packages/esm-patient-chart-app/src/visit/visits-widget/visit-detail-overview.test.tsx [hunk 4]
```diff
@@ -28,6 +29,7 @@ jest.mock('@openmrs/esm-framework', () => {
 describe('VisitDetailOverview', () => {
   it('renders an empty state view if encounters data is unavailable', async () => {
     mockOpenmrsFetch.mockReturnValueOnce({ data: { results: [] } });
+    mockGetConfig.mockReturnValue(Promise.resolve({ htmlFormEntryForms: [] }));
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/visit/visits-widget/visit-detail-overview.test.tsx:32]:
```suggestion
    mockGetConfig.mockResolvedValue({ htmlFormEntryForms: [] });
```

File: packages/esm-patient-chart-app/src/visit/visits-widget/visit-detail-overview.test.tsx [hunk 5]
```diff
@@ -63,6 +65,7 @@ describe('VisitDetailOverview', () => {
     const user = userEvent.setup();
 
     mockOpenmrsFetch.mockReturnValueOnce(visitOverviewDetailMockData);
+    mockGetConfig.mockReturnValue(Promise.resolve({ htmlFormEntryForms: [] }));
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/visit/visits-widget/visit-detail-overview.test.tsx:68]:
```suggestion
     mockGetConfig.mockResolvedValue({ htmlFormEntryForms: [] });
```

**Review by ibacher — APPROVED** [review]

**Comment by ibacher** [comment]
Another solution I don't love, but until we have something better, I don't think we have any choice but introducing this level of coupling.

**Pull request merged**